### PR TITLE
Fix read replica config option

### PR DESCRIPTION
**Pull request by kulkarnisourabh2**
- To fix following error in pipeline,

``` Error: Invalid value for module argument
  on .terraform/modules/stg_infra.postgres_db/main.tf line 96, in module "google_postgres_db":
  96:   read_replicas = [
  97:     for array_index in range(var.read_replica_count) : {
  98:       name = array_index
  99:       tier = var.instance_size_read_replica
 100:       zone = local.zone_read_replica
 101:       ip_configuration = {
 102:         authorized_networks = local.read_replica_authorized_networks
 103:         ipv4_enabled        = var.public_access_read_replica
 104:         private_network     = var.private_network
 105:         require_ssl         = null
 106:       }
 107:       database_flags  = local.db_flags_read_replica
 108:       disk_autoresize = var.disk_auto_resize_read_replica
 109:       disk_size       = var.disk_size_gb_read_replica
 110:       disk_type       = "PD_SSD"
 111:       user_labels     = var.labels_read_replica
 112:     }
 113:   ]
The given value is not suitable for child module variable "read_replicas"
defined at
.terraform/modules/stg_infra.postgres_db.google_postgres_db/modules/postgresql/variables.tf:174,1-25:
element 0: attribute "encryption_key_name" is required.```

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -290,3 +290,9 @@ variable "insights_config" {
   })
   default = null
 }
+
+variable "encryption_key_name" {
+  description = "Encryption key is required for replica in different region. For replica in same region as master set encryption_key_name = null"
+  type        = string
+  default     = "null"
```
Comment (syedrakib) [inline variables.tf:297]:
```suggestion
  description = "Encryption key is required for replicas in different regions. For replicas in same region as master, set encryption_key_name = null"
  type        = string
  default     = null
```

**Review by syedrakib — CHANGES_REQUESTED** [review]
Need to remove the quotes around `null` for default value.

File: main.tf [hunk 2]
```diff
@@ -95,9 +95,10 @@ module "google_postgres_db" {
   read_replica_name_suffix         = local.read_replica_name_suffix
   read_replicas = [
     for array_index in range(var.read_replica_count) : {
-      name = array_index
-      tier = var.instance_size_read_replica
-      zone = local.zone_read_replica
+      name                = array_index
+      tier                = var.instance_size_read_replica
+      zone                = local.zone_read_replica
+      encryption_key_name = var.encryption_key_name
```
Comment (syedrakib) [inline main.tf:101]:
```suggestion
      encryption_key_name = var.encryption_key_name_read_replica
```

File: variables.tf [hunk 3]
```diff
@@ -290,3 +290,9 @@ variable "insights_config" {
   })
   default = null
 }
+
+variable "encryption_key_name" {
```
Comment (syedrakib) [inline variables.tf:294]:
```suggestion
variable "encryption_key_name_read_replica" {
```

**Review by syedrakib — COMMENTED** [review]
adding suffix to the var to be consistent with rest of the read_replica vars

**Review by syedrakib — APPROVED** [review]

**Review by prajithparammal — APPROVED** [review]

**Pull request merged**